code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -624,13 +624,18 @@
 	desc = "A special box for sensitive people."
 	icon_state = "hugbox"
 	foldable = null
+	var/cooldown = 0
 
 /obj/item/weapon/storage/box/hug/attack_self(mob/user)
-	..()
-	user.changeNext_move(CLICK_CD_MELEE)

```

Comment:
@freerealestate Don't bother, the hotkey system we use is byond's inbuilt one, which points to a bunch of nonorganised procs that all fail to handle cooldowns.

I'm sure you'd probably be able to work it out for attack_self (this proc), but that's just a band aid fix.
When @Ikarrus suggested you fix cooldowns, I don't think they realised how messed up it currently is.

tl;dr it's going to be annoying as fuck, so feel free to not do it.